react/components/PickupBusinessHours.tsx

```diff
@@ -0,0 +1,183 @@
+import React from 'react'
+import type { InjectedIntlProps } from 'react-intl'
+import { injectIntl } from 'react-intl'
+
+import styles from './PickupBusinessHours.css'
+import { translate } from '../utils/i18nUtils'
+import PickupPointHour from './PickupPointHour'
+
+const DAYS_ORDER = [1, 2, 3, 4, 5, 6, 0]
+
+const MONDAY = 1
+const FRIDAY = 5
+
+const MONDAY_INDEX = DAYS_ORDER.indexOf(MONDAY)
+const FRIDAY_INDEX = DAYS_ORDER.indexOf(FRIDAY)
+
+function doesWeekDaysHaveTheSameHours(businessHours: NormalizedBusinessHour[]) {
+  const weekDays = businessHours.slice(MONDAY_INDEX, FRIDAY_INDEX + 1)
+
+  const [monday] = weekDays
+
+  let previousOpeningTime = monday.closed ? null : monday.openingTime
+  let previousClosingTime = monday.closed ? null : monday.closingTime
+
+  for (let i = 1; i < weekDays.length; i++) {
+    const currentDay = weekDays[i]
+
+    const currentDayOpeningTime = currentDay.closed
+      ? null
+      : currentDay.openingTime
+
+    const currentDayClosingTime = currentDay.closed
+      ? null
+      : currentDay.closingTime
+
+    if (
+      currentDayOpeningTime !== previousOpeningTime ||
+      currentDayClosingTime !== previousClosingTime
+    ) {
+      return false
+    }
+
+    previousOpeningTime = currentDayOpeningTime
+    previousClosingTime = currentDayClosingTime
+  }
+
+  return true
+}
+
+function condenseWeekDaysHours(businessHours: NormalizedBusinessHour[]) {
+  return businessHours.reduce<NormalizedBusinessHour[]>(
+    (acc, businessHour, index) => {
+      if (index >= MONDAY_INDEX && index <= FRIDAY_INDEX) {
+        if (index === MONDAY_INDEX) {
+          return acc.concat({
+            number: '1to5',
+            ...(businessHour.closed
+              ? {
+                  closed: businessHour.closed,
+                }
+              : {
+                  closed: businessHour.closed,
+                  openingTime: businessHour.openingTime,
+                  closingTime: businessHour.closingTime,
+                }),
+          })
+        }
+
+        return acc
+      }
+
+      return acc.concat(businessHour)
+    },
+    []
+  )
+}
+
+type NormalizedBusinessHour =
+  | {
+      closed: true
+      number: string | number
+    }
+  | {
+      openingTime: string
+      closingTime: string
+      closed: false
+      number: string | number
+    }
+
+function normalizeBusinessHours(
+  businessHours: BusinessHour[]
+): NormalizedBusinessHour[] {
+  return DAYS_ORDER.map((number) => {
+    const day = businessHours.find(
+      (businessDay) => number === businessDay.DayOfWeek
+    )
+
+    const dayObject = day && {
+      closed: false,
+      openingTime: day.OpeningTime,
+      closingTime: day.ClosingTime,
+    }
+
+    return {
+      number,
+      closed: true,
+      ...(dayObject ?? {}),
+    }
+  })
```

Comment:
This is returning an error in the linter 🤔 . I managed to make it work with

```tsx
  return DAYS_ORDER.map((number) => {
    const day = businessHours.find(
      (businessDay) => number === businessDay.DayOfWeek
    )

    if (!day) {
      return {
        number,
        closed: true,
      }
    }

    return {
      number,
      closed: false,
      openingTime: day.OpeningTime,
      closingTime: day.ClosingTime,
    }
  })
```

Reply:
can you show the error? I don't see it here :-/

Reply:
for some reason it shows an error in VSCode, but if you run `yarn lint:ts` in the repo root it doesn't show anything 🤔

Reply:
fixed :-)

Reply:
hmm so I guess this is a misconfiguration on my side?

Reply:
I don't think so, I was able to see the issue but only in vscode. Maybe it is using a different TS version that throws an error on this case